app/pipeline/runner/runner.go

```diff
@@ -45,7 +45,7 @@ func NewExecutionRunner(
 ) (*runtime2.Runner, error) {
 	// For linux, containers need to have extra hosts set in order to interact with
 	// the gitness container.
-	extraHosts := []string{"host.docker.internal:host-gateway"}
+	extraHosts := []string{config.URL.Container + ":host-gateway"}
```

Comment:
Hi @ppphp , this will break since `config.URL.Container` is a full URL (eg `http://host.docker.internal:3000`) and not `host.docker.internal`. You can check `https://github.com/harness/gitness/blob/main/cli/server/config.go#L89` this is where we create the container URL.
We should only have `host.docker.internal` or `host.containers.internal` here

Reply:
you can add a new method to the `url.Provider` (similar to `GetAPIHostname() string`) where you return the container host.

Reply:
thank you for your advice, but I pushed a simple way. If new commit is wrong, please close it~